dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/uriInfo/src/com/ibm/ws/jaxrs/fat/uriInfo/TestClientRequestResponseFilter.java

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs.fat.uriInfo;
+
+import java.io.IOException;
+
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.client.ClientResponseContext;
+import javax.ws.rs.client.ClientResponseFilter;
+
+public class TestClientRequestResponseFilter implements ClientRequestFilter, ClientResponseFilter {
+
+    private static String pre;
+
+    @Override
+    public void filter(ClientRequestContext requestContext) throws IOException {
+        pre = "client pre=" + requestContext.getUri().toString();
```

Comment:
I hadn't thought about testing the URI from the request context - I was thinking that we would test the same thing that is used in the resource class:
```
@Context
UriInfo uriInfo;
```

Could you swap to using the `UriInfo` object?  It might be nice to keep the existing test that the request and response context's URI is correct too.

Reply:
done, made a 2nd test with a 2nd filter